javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/TypeExtractor.java

```diff
@@ -54,11 +54,17 @@
 
 import java.util.List;
 import java.util.Optional;
+import java.util.function.Predicate;
 
 import static com.github.javaparser.resolution.Navigator.demandParentNode;
 
 public class TypeExtractor extends DefaultVisitorAdapter {
 
+    /**
+     * Returns {@code true} when the Node to be tested is not an
+     * {@link EnclosedExpr}, {@code false} otherwise.
+     */
+    private static final Predicate<Node> IS_NOT_ENCLOSED_EXPR = n -> !(n instanceof EnclosedExpr);
```

Comment:
if n is an expression you can write n.isEnclosedExpr() instead of using instanceof.

Reply:
This won't work.

Assume I change the implementation to 
```
private static final Predicate<Expression> IS_NOT_ENCLOSED_EXPR = n -> !(n.isEnclosedExpr());
```
As the Predicate is used as a parameter to the new `demandParentNode(Node, Predicate<Node>)` it would require me to change that method to `demandParentNode(Node, Predicate<Expression>)`. However in the implementation of `demandParentNode` we assume the result of a `demandParentNode` call is tested against the predicate , and that result is a `Node`.

Or did I missed something.